src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/WebInputSources.cs

```diff
@@ -12,13 +16,72 @@ internal static class WebInputSources
         /// </summary>
         public static ImmutableHashSet<SourceInfo> SourceInfos { get; }
 
+        /// <summary>
+        /// Cached information if the specified symbol is a Asp.Net Core Controller: (compilation) -> ((class symbol) -> (is Controller))
+        /// </summary>
+        private static readonly BoundedCacheWithFactory<Compilation, ConcurrentDictionary<ISymbol, bool>> s_classIsControllerByCompilation =
```

Comment:
Nit: Could make the type more specific, since we're already checking if `methodSymbol.ContainingType` is an `INamedTypeSymbol`:

```suggestion
        private static readonly BoundedCacheWithFactory<Compilation, ConcurrentDictionary<INamedTypeSymbol, bool>> s_classIsControllerByCompilation =
```